autoapi/templates/python/module.rst

```diff
@@ -10,10 +10,65 @@
 
 {% endif %}
 
+{% block subpackages %}{% if obj.subpackages %}
+Subpackages
+-----------
+.. toctree::
+   :maxdepth: 3
+{% for subpackage in obj.subpackages %}
+   {{ subpackage.short_name }}/index.rst
+{%- endfor %}
+{% endif %}{% endblock %}
+
+{% block submodules %}{% if obj.submodules %}
+Submodules
+----------
+.. toctree::
+   :maxdepth: 1
+{% for submodule in obj.submodules %}
+   {{ submodule.short_name }}/index.rst
+{%- endfor %}
+{% endif %}{% endblock %}
+
+
 {% block content %}
+{{ obj.type|title }} Contents
+{{ "-" * obj.type|length }}---------
+
+{% block classes %}{% if obj.classes %}
+Classes
+~~~~~~~
+
+.. autoapisummary::
+{% for klass in obj.classes %}
+   {{ klass.id }}
+{%- endfor %}
+{% endif %}{% endblock %}
+
+{% block methods %}{% if obj.methods %}
+Methods
+~~~~~~~
+
+.. autoapisummary::
+
+{% for method in obj.methods %}
+   {{ method.id }}
+{%- endfor %}
+{% endif %}{% endblock %}
+
+{% block functions %}{% if obj.functions %}
+Functions
+~~~~~~~~~
+.. autoapisummary::
+
+{% for function in obj.functions %}
+   {{ function.id }}
+{%- endfor %}
+{% endif %}{% endblock %}
+
```

Comment:
If we make autoapisummary optional, templates will need to change to be conditional as well.